anitya/tests/lib/versions/test_python.py

```diff
@@ -283,3 +290,79 @@ def test_normalization_ws(self):
         version = python.PythonVersion(version=" 1.0.0\n")
         self.assertEqual(str(version), "1.0.0")
         self.assertEqual(version, python.PythonVersion(version="1.0.0"))
+
+    def test_sorting(self):
+        """
+        Assert that https://github.com/fedora-infra/anitya/issues/1586
+        is fixed.
+        """
+        versions = [
+            "4.0.0",
+            "4.0.0rc1",
+            "4.0.0rc0",
+            "4.0.0b2",
+            "4.0.0b1",
+            "4.0.0b0",
+            "4.0.0a9",
+            "4.0.0a8",
+            "4.0.0a7",
+            "4.0.0a6",
+            "4.0.0a4",
+            "4.0.0a37",
+            "4.0.0a36",
+            "4.0.0a35",
+            "4.0.0a34",
+        ]
+        python_versions = []
+        for version in versions:
+            python_versions.append(python.PythonVersion(version))
```

Comment:
This might be a tad easier to read:

```suggestion
        python_versions = [python.PythonVersion(v) for v in versions]
```